v0/destinations/google_cloud_function/transform.js

```diff
@@ -0,0 +1,156 @@
+const {
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  getSuccessRespEvents,
+  getErrorRespEvents,
+  CustomError
+
+} = require("../../util");
+const {
+  TRIGGERTYPE
+} = require("./config");
+
+const {
+  generateBatchedPayloadForArray
+} = require("./util");
+
+function process(event) {
+  const {
+      message,
+      destination
+  } = event;
+  const {
+      Config
+  } = destination;
+  const {
+      googleCloudFunctionUrl,
+      triggerType,
+      apiKeyId,
+      gcloudAuthorization
+  } = Config;
+  const basicAuth = Buffer.from(`apiKey:${apiKeyId}`).toString("base64");
+  if (!googleCloudFunctionUrl) {
+      throw new CustomError("Cloud Function Url not found. Aborting", 400);
+  }
+  const response = defaultRequestConfig();
+  if (TRIGGERTYPE.HTTPS === triggerType) {
+      response.headers = {
+          "content-type": "application/json",
+          Authorization: `Bearer ${gcloudAuthorization}`,
+          ApiKey: `Basic ${basicAuth}`
+      };
+  } else {
+      response.headers = {
+          "content-type": "application/json",
+          Authorization: `Basic ${basicAuth}`
+      };
+  }
```

Comment:
We are already adding this in the batching function. So we can remove it from here.

Reply:
added common code in util file